### ISSUE TITLE
[bug] set-state-in-effect: render-phase set reported (once per effect) when an effect calls a prop function

### ISSUE THREAD

**Issue by jsnkle** [Rel1cx/eslint-react#1944]
### Describe the bug

`react-x/set-state-in-effect` reports a **render-phase** `set` call — the "adjusting state when a prop changes" pattern from the React docs, which is not inside any effect — as "called synchronously in an effect". The false positive only appears when the component also has a `useEffect` whose body calls a **prop function**, and it is reported **once per such effect**: two effects → the same render-phase line reported twice.

Effects whose body calls `console.log`, a module-scope function, or a `useCallback` local do not trigger it, and neither does wrapping the render-phase set in a helper function — so the report is tied to the effect calling a prop, not to the set call itself.

### Reproduction

`One.tsx` — one warning, at line 5 (the render-phase `setPage`, not inside the effect):

```tsx
import { useEffect, useState } from "react";

export function One({ target, onDone }: { target: number; onDone: () => void }) {
  const [page, setPage] = useState(1);
  if (page !== target) setPage(target); // ← 5:24 "Do not call the 'set' function 'setPage' of 'useState' synchronously in an effect"
  useEffect(() => {
    onDone();
  }, [onDone, target]);
  return <div>{page}</div>;
}
```

`Two.tsx` — same as above plus a second effect calling `onDone()` → **the same line 5:24 is reported twice**.

```tsx
import { useEffect, useState } from "react";

export function Two({ target, onDone }: { target: number; onDone: () => void }) {
  const [page, setPage] = useState(1);
  if (page !== target) setPage(target); // ← reported at 5:24, twice
  useEffect(() => {
    onDone();
  }, [onDone, target]);
  useEffect(() => {
    onDone();
  }, [onDone]);
  return <div>{page}</div>;
}
```

`Control.tsx` — identical shape, but the effect calls `console.log(target)` instead of a prop → **no warning**.

```tsx
import { useEffect, useState } from "react";

export function Control({ target }: { target: number }) {
  const [page, setPage] = useState(1);
  if (page !== target) setPage(target);
  useEffect(() => {
    console.log(target);
  }, [target]);
  return <div>{page}</div>;
}
```

Output (`eslint src/__repro`, flat config, rule at `warn`):

```
src/__repro/One.tsx
  5:24  warning  Do not call the 'set' function 'setPage' of 'useState' synchronously in an effect. ...  react-x/set-state-in-effect
src/__repro/Two.tsx
  5:24  warning  Do not call the 'set' function 'setPage' of 'useState' synchronously in an effect. ...  react-x/set-state-in-effect
  5:24  warning  Do not call the 'set' function 'setPage' of 'useState' synchronously in an effect. ...  react-x/set-state-in-effect
✖ 3 problems (0 errors, 3 warnings)
```

Also verified: replacing `onDone()` with a module-scope function or a `useCallback` local clears it; moving `if (page !== target) setPage(target)` into a small helper function (`clampDuringRender(page, target, setPage)`) clears it too.

### Expected behavior

No report for `One.tsx` / `Two.tsx`: the `setPage` call is in the render phase, not in an effect. If the rule does intend to flag something here, it should at least report each location once.

### Platform and versions

- eslint-plugin-react-x 5.18.3 (also seen on 5.18.1)
- eslint 10.8.1
- typescript 5.9.3, @typescript-eslint parser (type-aware config)
- react 19.2.8
- Node 24.19.0, macOS

### Additional context

Found while clearing `set-state-in-effect` warnings in a larger component: applying the render-phase conversion to a component that had two effects calling prop callbacks turned 3 real warnings into 6 (the 3 converted sites, each reported twice). Wrapping the render-phase sets in helper functions is a workable sidestep, which is presumably why the pattern often lints clean in the wild. Happy to add a test case PR if you can point me at the right spec file.
